java/src/server/client/GameServer.java

```diff
@@ -53,7 +56,9 @@ public UserSession register(@NotNull Credentials credentials) throws CredentialN
 
     @Override
     public GameInfo[] listOfGames() throws IllegalArgumentException, CommunicationException {
-        return new GameInfo[0];
+        ArrayList<GameInfo> gameList = getServerModel().getGameInfo();
```

Comment:
This is causing a compilation error. It should be receiving a `List`.
A rule of thumb is to always have variables and methods receive and return interfaces, and not implementations.